elastic-buffer-sim/src/Bittide/ClockControl/Strategies/Callisto.hs

```diff
@@ -2,81 +2,130 @@
 --
 -- SPDX-License-Identifier: Apache-2.0
 
-{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
 
 module Bittide.ClockControl.Strategies.Callisto
   ( callisto
   )
 where
 
-import Clash.Explicit.Prelude
+import Clash.Prelude
 
 import Bittide.ClockControl
 
+import qualified Clash.Signal.Delayed as D
+import qualified Clash.Cores.Xilinx.Floating as F
+
 data ControlSt = ControlSt
-  { x_k :: Float -- ^ 'x_k' is the integral of the measurement
-  , z_k :: Integer
-  , b_k :: SpeedChange
+  { _x_k :: !Float -- ^ 'x_k' is the integral of the measurement
+  , _z_k :: !(Signed 32)
+  , _b_k :: !SpeedChange
   } deriving (Generic, NFDataX)
 
-initControlSt :: ControlSt
-initControlSt = ControlSt 0 0 NoChange
+-- | Initial state of control
+initState :: ControlSt
+initState = ControlSt 0 0 NoChange
+
+-- | A counter that starts at a given value, counts down, and if it reaches
+-- zero wraps around to the initial value.
+wrappingCounter ::
+  (HiddenClockResetEnable dom, KnownNat n) =>
+  Unsigned n ->
+  Signal dom (Unsigned n)
+wrappingCounter upper = counter
+ where
+  counter = register upper (go <$> counter)
 
+  go 0 = upper
+  go n = pred n
+
+-- | Clock correction strategy based on:
+--
+--   https://github.com/bittide/Callisto.jl
+--
 callisto ::
   forall n dom.
-  (KnownDomain dom, KnownNat n, 1 <= n) =>
-  Clock dom ->
-  Reset dom ->
-  Enable dom ->
-  ClockControlConfig ->
+  (HiddenClockResetEnable dom, KnownNat n, 1 <= n) =>
+  -- | Target data count. See 'targetDataCount'.
+  DataCount ->
+  -- | Provide an update every /n/ cycles
```

Comment:
You can use TeX in haddock!

https://haskell-haddock.readthedocs.io/en/latest/markup.html

Reply:
I'm going to move forward with the merge, given that this isn't critical. Please make a papercut PR if you feel I should have done a different mark up style :).